### ISSUE TITLE
gnt-backup import fails with IPs in a gnt-network

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#772]
Originally reported of Google Code with ID 716.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

<b>What distribution are you using?</b>
root@node1 ~ # cat /etc/debian_version
7.4
root@node1 ~ # gnt-cluster --version
gnt-cluster (ganeti v2.9.3) 2.9.3
root@node1 ~ # gnt-cluster version
Software version: 2.9.3
Internode protocol: 2090000
Configuration format: 2090000
OS api version: 20
Export interface: 0
VCS version: v2.9.3
root@node1 ~ # hspace --version
hspace (ganeti) version v2.9.3
compiled with ghc 7.4
running on linux x86_64


<b>What steps will reproduce the problem?</b>
1. root@node1 /usr/share/ganeti/os/raw # gnt-backup import -t plain -n node1 test2.30c3.tailored-vm.de  
Failure: prerequisites not met for this operation:
error type: wrong_state, error details:
The requested IP address (10.213.100.21) belongs to network publicNAT, but the target NIC does not.

<b>What is the expected output? What do you see instead?</b>

that the VM is imported; the error


<b>Please provide any additional information below.</b>

root@node1 ~ # gnt-network info publicNAT
Network name: publicNAT
UUID: d3f690eb-4b05-4500-9a2e-321274e32e79
Serial number: 1
  Subnet: 10.213.0.0/16
[...]

root@node1 ~ # cat /var/lib/ganeti/export/test2.30c3.tailored-vm.de/config.ini
[...]
[instance]
name = test2.30c3.tailored-vm.de
[...]
nic0_ip = 10.213.100.21
nic0_network = d3f690eb-4b05-4500-9a2e-321274e32e79
nic0_link = br0
nic0_mode = bridged
[...]



```
Originally added on 2014-02-13 02:54:38 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This bug exists in 2.8 too! I fix in on 2.8 and then it will need a merge/cherry-pick to the later branches.

BTW as I see NIC name is not exported/imported at all! Should I fix this too?
I guess the same applies to Disks too.
```
Originally added on 2014-02-13 08:14:06 +0000 UTC.

Added Labels: Type-Defect Priority-Critical Component-networking Component-instances
Changed State: Accepted
Added to Milestone: Release2.8

**Comment by Ganeti-Issues-Migrator** [comment]
```
Yes, please fix the export of the nic name (and possibly disk name) as well while you are at it. Thanks a lot!
```
Originally added on 2014-02-13 08:34:41 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 0f68f7fab33fd6e09758d59bcd168266d957311f
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Thu Feb 13 16:47:01 2014 +0200

    Export and import Disk/NIC name
    
    Name of Disk/NIC were not exported during backup until now.
    Use the exported info during gnt-backup import.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Signed-off-by: Michele Tartara <mtartara@google.com>
    Reviewed-by: Michele Tartara <mtartara@google.com>

commit 0973f9ed9864f1e56e160676f2181bbe14c2f925
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Thu Feb 13 16:47:00 2014 +0200

    Fix backup import in case NIC is inside a network
    
    Network UUID is written in .ini file during backup export
    but is not used by _ReadExportParams(). This patch fixes it.
    
    Please note that in case a network is given, link and mode should
    not be included in NIC options.
    
    This fixes issue 716.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Signed-off-by: Michele Tartara <mtartara@google.com>
    Reviewed-by: Michele Tartara <mtartara@google.com>

commit 523170de667f8c9f02393fe4b16aa2a9c9226a81
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Thu Feb 13 16:46:59 2014 +0200

    Override get() method of ConfigParser
    
    During backup import/export SafeConfigParser() is used to
    save/restore instance's configuration. There is a possibility if an
    export is done with a different Ganeti version, a specific value not
    to be saved during export (e.g. the NIC/Disk name) but still
    requested during import.
    
    With this patch we override the get() method of SafeConfigParser()
    and catch NoOptionError if raised and return None. Additionally we
    translate "None" values read from .ini file into None.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Signed-off-by: Michele Tartara <mtartara@google.com>
    Reviewed-by: Michele Tartara <mtartara@google.com>

```
Originally added on 2014-02-20 14:48:24 +0000 UTC.

Mentioned Issue with GoogleCode ID 716 has been migrated to GitHub with ID #772.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released in the 2.9.5 released.
```
Originally added on 2014-02-27 11:58:33 +0000 UTC.

Changed State: Released
